src/gsi/m_berror_stats_reg.f90

```diff
@@ -807,103 +819,126 @@ subroutine berror_read_wgt_reg(msig,mlat,corz,corp,hwll,hwllp,vz,rlsig,varq,qopt
-           hwllp(i,n0+n)=hwll(i,1,nrf3_q)
+           hwllp(i,loc)=hwll(i,1,nrf3_q)
         end do
-     endif
 
-     if (cvarsmd(n)=='gustwter') then
+     else if (cvarsmd(n)=='gustwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_gust)
+           corp(i,loc)=corp(i,nrf2_gust)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_gust)
+           hwllp(i,loc)=hwllp(i,nrf2_gust)
         end do
-     endif
 
-     if (cvarsmd(n)=='wspd10mwter') then
+     else if (cvarsmd(n)=='wspd10mwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_wspd10m)
+           corp(i,loc)=corp(i,nrf2_wspd10m)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_wspd10m)
+           hwllp(i,loc)=hwllp(i,nrf2_wspd10m)
         end do
-     endif
 
-     if (cvarsmd(n)=='td2mwter') then
+     else if (cvarsmd(n)=='td2mwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_td2m)
+           corp(i,loc)=corp(i,nrf2_td2m)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_td2m)
+           hwllp(i,loc)=hwllp(i,nrf2_td2m)
         end do
-     endif
 
-     if (cvarsmd(n)=='mxtmwter') then
+     else if (cvarsmd(n)=='mxtmwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_mxtm)
+           corp(i,loc)=corp(i,nrf2_mxtm)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_mxtm)
+           hwllp(i,loc)=hwllp(i,nrf2_mxtm)
         end do
-     endif
 
-     if (cvarsmd(n)=='mitmwter') then
+     else if (cvarsmd(n)=='mitmwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_mitm)
+           corp(i,loc)=corp(i,nrf2_mitm)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_mitm)
+           hwllp(i,loc)=hwllp(i,nrf2_mitm)
         end do
-     endif
 
-     if (cvarsmd(n)=='uwnd10mwter') then
+     else if (cvarsmd(n)=='uwnd10mwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_uwnd10m)
+           corp(i,loc)=corp(i,nrf2_uwnd10m)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_uwnd10m)
+           hwllp(i,loc)=hwllp(i,nrf2_uwnd10m)
         end do
-     endif
 
-     if (cvarsmd(n)=='vwnd10mwter') then
+     else if (cvarsmd(n)=='vwnd10mwter') then
         do i=1,mlat
-           corp(i,n0+n)=corp(i,nrf2_vwnd10m)
+           corp(i,loc)=corp(i,nrf2_vwnd10m)
         end do
         do i=0,mlat+1
-           hwllp(i,n0+n)=hwllp(i,nrf2_vwnd10m)
+           hwllp(i,loc)=hwllp(i,nrf2_vwnd10m)
         end do
-     endif
+!  if not found use default
+     else 
+        do i=1,mlat
+           corp(i,loc)=corz_default
+        end do
+        do i=0,mlat+1
+           hwllp(i,loc)=hwll_default
+        end do
+     end if
   enddo
 
   deallocate(nrf3_loc,nrf2_loc,nmotl_loc)
+! Normalize vz with del sigmma and convert to vertical grid units!
+  if( .not. usenewgfsberror)then
+     dlsig(1)=rlsig(1)-rlsig(2)
```

Comment:
Mike,

I have the suggested changes, plus some small comment changes that were suggested by the reviewers.  I am afraid that I may have messed up the commit message because of some earlier changes that I made.

Johhn

Reply:
Please note that as Catherine noted we should include the version of the global berror stats containing the extra record.  Not sure what a good name would be.  Maybe include latsigma in the name.

Reply:
Hi John,

If you have committed changes that you have previously made without using:

`git commit --amend`

you can correct this using:

`git reset --soft HEAD~1`

This command will undo your previous commit, but keep the changes that were made.

You can use this command until you are back to the original commit, then use:

`git commit --amend`

If you changed the commit message using:

`git commit --amend`

You don't need to worry about it, so long as the GitHub issue number is present at the start of the commit message.

Please let me know if you have any questions.

Reply:
Mike,

I  guess I was doing things even more wrong.  I made my changes to the branch rather than my master.  I have now made the changes to the master and done the commit properly (I think).  I am rerunning test cases to ensure everything is OK with global (where I made some minor changes).  Will note here when completed and OK.

John

Reply:
Everything appears OK.